packages/repository/src/__tests__/unit/repositories/relations-helpers/mongo-related-helpers.unit.ts

```diff
@@ -0,0 +1,139 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/repository-tests
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect} from '@loopback/testlab';
+import {ObjectID} from 'bson';
+import {belongsTo, Entity, hasMany, model, property} from '../../../..';
+import {
+  isBsonType,
+  deduplicate,
+  normalizeKey,
+  buildLookupMap,
+  reduceAsArray,
+} from '../../../../relations';
+
+describe('unit tests, simulates mongodb env for helpers of ininclusion resolver ', () => {
+  describe('helpers for formating instances', async () => {
+    it('isBsonType', async () => {
+      const id1 = new ObjectID();
+      const id2 = new ObjectID();
+      expect(isBsonType(id1)).to.be.true();
+      expect(isBsonType(id2)).to.be.true();
+    });
+    it('simple deduplcate + isBsonType', async () => {
+      const id1 = new ObjectID();
+      const id2 = new ObjectID();
+
+      const result = deduplicate([id1, id2]);
+
+      expect(result).to.deepEqual([id1, id2]);
+    });
+    it('multiple items deduplcate + isBsonType', async () => {
+      const id1 = new ObjectID();
+      const id2 = new ObjectID();
+      const id3 = new ObjectID();
+      expect(isBsonType(id1)).to.be.true();
+
+      const result = deduplicate([id3, id1, id1, id3, id2]);
+
+      expect(result).to.deepEqual([id3, id1, id2]);
+    });
+  });
+
+  describe('helpers for generating inclusion resolvers', async () => {
+    it('normalizeKey', async () => {
+      const id = new ObjectID();
+      expect(normalizeKey(id)).to.eql(id.toHexString());
+    });
+    // the tests below simulate mongodb environment.
+    it('simple normalizeKey + buildLookupMap', async () => {
+      const id = new ObjectID();
+      expect(normalizeKey(id)).to.eql(id.toHexString());
+    });
+    it('normalizeKey + buildLookupMap: returns multiple instances in an array', async () => {
+      const categoryId = new ObjectID();
+      const anotherCategoryId = new ObjectID();
+      const pen = createPen(categoryId);
+      const pencil = createPencil(categoryId);
+      createPEraser(anotherCategoryId);
+
+      const result = buildLookupMap<unknown, Product, Category[]>(
+        [pen, pencil],
+        'categoryId',
+        reduceAsArray,
+      );
+      // expects this map to have String/Product pair
+      const expected = new Map<String, Array<Product>>();
+      const strId = categoryId.toString();
+      expected.set(strId, [pen, pencil]);
+      expect(result).to.eql(expected);
+    });
+    it('normalizeKey + buildLookupMap: return instances in multiple arrays', async () => {
```

Comment:
When I read the test names, I find it difficult to understand what' the difference between this test (return instances in multiple arrays) and the previous one (return instances in multiple arrays). Can you please find more descriptive names?

Also if you have a group of tests sharing the same context (normalizeKey + buildLookupMap), then we usually put them in a `context` block.

```ts
context('normalizeKey + buildLookupMap', () => {
  it('returns multiple instances in an array', async () => {
    // ...
  });
  // ...
});
```

Last but not least - if your test code is synchronous (no `await` calls), then it's better to use a regular function `() => {}` instead of `async () => {}`.